src/pages/[platform]/build-a-backend/auth/advanced-workflows/index.mdx

```diff
@@ -0,0 +1,789 @@
+      // You can also refresh the token if it's expired before proceeding
+
+      const getIdResult = await cognitoidentity.getId({
+        // Get the identityPoolId from config
+        IdentityPoolId: '<identity-pool-id-from-config>',
+        Logins: { [this.federatedLogin.domain]: this.federatedLogin.token },
+      });
+
+      const cognitoCredentialsResult = await cognitoidentity.getCredentialsForIdentity({
+        IdentityId: getIdResult.IdentityId,
+        Logins: { [this.federatedLogin.domain]: this.federatedLogin.token },
+      });
+
+      const credentials: CredentialsAndIdentityId = {
+        credentials: {
+          accessKeyId: cognitoCredentialsResult.Credentials?.AccessKeyId,
+          secretAccessKey: cognitoCredentialsResult.Credentials?.SecretKey,
+          sessionToken: cognitoCredentialsResult.Credentials?.SessionToken,
+          expiration: cognitoCredentialsResult.Credentials?.Expiration,
+        },
+        identityId: getIdResult.IdentityId,
+      };
+      return credentials;
+    } catch (e) {
+      console.log('Error getting credentials: ', e);
+    }
+  }
+  // Implement this to clear any cached credentials and identityId. This can be called when signing out of the federation service.
+  clearCredentialsAndIdentityId(): void {}
+}
+
+// Create an instance of your custom provider
+const customCredentialsProvider = new CustomCredentialsProvider();
+Amplify.configure(awsconfig, {
+  Auth: {
+    // Supply the custom credentials provider to Amplify
+    credentialsProvider: customCredentialsProvider
+  },
+});
+
+```
+
+Now that the custom credentials provider is built and supplied to `Amplify.configure`, let's look at how you can use the custom credentials provider to finish federation into Cognito identity pool.
+
+<InlineFilter filters={['react-native']}>
+
+### Facebook Sign-in (React Native - Expo)
+
+```javascript
+import Expo from 'expo';
+import React from 'react';
+import { fetchAuthSession } from 'aws-amplify/auth';
+
+const App = () => {
+  const signIn = async () => {
+    const { type, token, expires } =
+      await Expo.Facebook.logInWithReadPermissionsAsync(
+        'YOUR_FACEBOOK_APP_ID',
+        {
+          permissions: ['public_profile']
+        }
+      );
+    if (type === 'success') {
+      // sign in with federated identity
+      try {
+        customCredentialsProvider.loadFederatedLogin({
+          domain: 'graph.facebook.com',
+          token: token
+        });
+        const fetchSessionResult = await fetchAuthSession(); // will return the credentials
+        console.log('fetchSessionResult: ', fetchSessionResult);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+  };
+
+  // ...
+
+  return (
+    <View style={styles.container}>
+      <Button title="FBSignIn" onPress={signIn} />
+    </View>
+  );
+};
+```
+
+</InlineFilter>
+
+<InlineFilter filters={['javascript', "angular", "nextjs", "react", "vue"]}>
```

Comment:
Should `vue` and angular be removed from this since the inline content is for `react`?

I'm also wondering if `javascript` should be removed too?